testing/tests/DevExpress.ui.widgets/treeViewParts/checkboxes.js

```diff
@@ -238,3 +239,21 @@ QUnit.test("Selection works correct with custom rootValue", function(assert) {
     assert.ok(nodes[0].items[0].selected, "item was selected");
     assert.strictEqual(nodes[0].selected, undefined, "item selection has undefined state");
 });
+
+
+['dataSource', 'items'].forEach((optionName) => {
+    QUnit.test(`Select item works even with loop/cycle in ${optionName} option`, function(assert) {
+        let options = { dataStructure: "plain", rootValue: 1, showCheckBoxesMode: "normal" };
+        options[optionName] = [
+            { id: 1, text: "item1", parentId: 2, selected: false, expanded: true },
+            { id: 2, text: "item1_1", parentId: 1, selected: false, expanded: true }];
+
+        const treeView = createInstance(options);
+
+        treeView.checkSelectedNodes([]);
+
+        const $parent = treeView.getElement().find('[aria-level="1"]');
+        treeView.instance.selectItem($parent);
```

Comment:
Do we need to check 'unselectItem'/'unselectAll'/'selectAll' with the same configs?

Reply:
Yes, I think so - great idea.
Add more test cases